File: app/src/main/java/org/teamblueridge/pasteitapp/UploadDownloadUrlPrep.java
package org.teamblueridge.pasteitapp;

import android.content.SharedPreferences;
import android.util.Log;

/**
 * Class to handle the create, recent, and languages URLs for the Stikked API</p>
 *
 * @author Kyle Laker (kalaker)
 * @version 1.0
 */
public class UploadDownloadUrlPrep {
    private static final String TAG = "TeamBlueRidge";
    private static final String TEAMBLUERIDGE_APIKEY = "teamblueridgepaste";
    private static final String LANGS_WITH_APIKEY = "/api/langs?apikey=";
    private static final String CREATE_WITH_APIKEY = "/api/create?apikey=";
    private static final String RECENT_WITH_APIKEY = "/api/recent?apikey=";
    private String mUrl;

    /**
     * Gets the proper URL for uploading a paste or for downloading a list of recent pastes (JSON)
     * or a list of languages the server supports (JSON) for syntax highlighting</p>
     *
     * @param prefs  The preferences to be able to access the domain, API key, etc.
     * @param upDown "upCreate" for creating a new paste
     *               "downLangs" for getting languages
     *               "downRecents" for recent pasts
     * @return The URL to be used, with the API key if necessary
     */
    public String prepUrl(SharedPreferences prefs, String upDown) {

        // Ensure that the paste URL is set, if not, default to Team BlueRidge
        String mPasteDomain;
        String mLangDownloadUrl;
        String mRecentDownloadUrl;
        String mUploadUrl;

        if (!prefs.getString("pref_domain", "").isEmpty()) {
            mPasteDomain = prefs.getString("pref_domain", "");
        } else {
            mPasteDomain = "https://paste.teamblueridge.org";
        }

        // Only set the API key for Team BlueRidge because we know our key
        if (mPasteDomain.equals("https://paste.teamblueridge.org")) {
            mUploadUrl = mPasteDomain + CREATE_WITH_APIKEY + TEAMBLUERIDGE_APIKEY;
            mLangDownloadUrl = mPasteDomain + LANGS_WITH_APIKEY + TEAMBLUERIDGE_APIKEY;
            mRecentDownloadUrl = mPasteDomain + RECENT_WITH_APIKEY + TEAMBLUERIDGE_APIKEY;
        } else {
            if (!prefs.getString("pref_api_key", "").isEmpty()) {
                String mPasteApiKey = prefs.getString("pref_api_key", "");
                mUploadUrl = mPasteDomain + CREATE_WITH_APIKEY + mPasteApiKey;
                mLangDownloadUrl = mPasteDomain + LANGS_WITH_APIKEY + mPasteApiKey;
                mRecentDownloadUrl = mPasteDomain + RECENT_WITH_APIKEY + mPasteApiKey;
            } else {
                mUploadUrl = mPasteDomain + "/api/create";
                mLangDownloadUrl = mPasteDomain + "/api/langs";
                mRecentDownloadUrl = mPasteDomain + "/api/recent";
            }
        }

        switch (upDown) {
            case "upCreate":
                mUrl = mUploadUrl;
                break;
            case "downLangs":
                mUrl = mLangDownloadUrl;
                break;
            case "downRecent":
                mUrl = mRecentDownloadUrl;
                break;
            default:
                Log.e(TAG, "Unknown URL case");
                break;
        }

        return mUrl;
    }
}

File: app/src/main/kotlin/org/teamblueridge/pasteitapp/UploadDownloadUrlPrep.kt
package org.teamblueridge.pasteitapp

import android.content.SharedPreferences
import android.util.Log

/**
 * Handles the URLs for uploading pastes, downloading languages, and downloading
 * recent pastes.
 *
 * @author Kyle Laker (kalaker)
 * @version 1.0
 */
class UploadDownloadUrlPrep {

    companion object {
        public val UPLOAD_PASTE = 1
        public val DOWNLOAD_LANGS = 2
        public val DOWNLOAD_RECENT = 3
        private val TAG = "TeamBlueRidge"
    }
    /**
     * Gets the proper URL for uploading a paste or for downloading a list of recent pastes (JSON)
     * or a list of languages the server supports (JSON) for syntax highlighting
     *
     * @param prefs   The preferences to be able to access the domain, API key, etc.
     * @param upDown  One of three options, depending on the API information needed
     *                <li><code>UPLOAD_PASTE</code>: creating a new paste
     *                <li><code>DOWNLOAD_LANGS</code>: getting languages
     *                <li><code>DOWNLOAD_RECENT</code>: recent pasts
     * @return The URL to be used, with the API key if necessary
     */
    fun prepUrl(prefs: SharedPreferences, upDown: Int): String {
        val TEAMBLUERIDGE_APIKEY = "teamblueridgepaste"
        val LANGS_WITH_APIKEY = "/api/langs?apikey="
        val CREATE_WITH_APIKEY = "/api/create?apikey="
        val RECENT_WITH_APIKEY = "/api/recent?apikey="

        val mLangDownloadUrl: String
        val mRecentDownloadUrl: String
        val mUploadUrl: String

        val mPasteDomain =
            if (!prefs.getString("pref_domain", "").isEmpty())
                prefs.getString("pref_domain", "")
            else
                "https://paste.teamblueridge.org"

        // Only set the API key for Team BlueRidge because we know our key
        if (mPasteDomain.equals("https://paste.teamblueridge.org")) {
            mUploadUrl = mPasteDomain + CREATE_WITH_APIKEY + TEAMBLUERIDGE_APIKEY
            mLangDownloadUrl = mPasteDomain + LANGS_WITH_APIKEY + TEAMBLUERIDGE_APIKEY
            mRecentDownloadUrl = mPasteDomain + RECENT_WITH_APIKEY + TEAMBLUERIDGE_APIKEY
        } else if (!prefs.getString("pref_api_key", "").isEmpty()) {
            val mPasteApiKey = prefs.getString("pref_api_key", "")
            mUploadUrl = mPasteDomain + CREATE_WITH_APIKEY + mPasteApiKey
            mLangDownloadUrl = mPasteDomain + LANGS_WITH_APIKEY + mPasteApiKey
            mRecentDownloadUrl = mPasteDomain + RECENT_WITH_APIKEY + mPasteApiKey
        } else {
            mUploadUrl = mPasteDomain + "/api/create"
            mLangDownloadUrl = mPasteDomain + "/api/langs"
            mRecentDownloadUrl = mPasteDomain + "/api/recent"
        }
        return when (upDown) {
            UPLOAD_PASTE    -> mUploadUrl
            DOWNLOAD_LANGS  -> mLangDownloadUrl
            DOWNLOAD_RECENT -> mRecentDownloadUrl
            else -> {
                Log.e(TAG, "Unknown URL case")
                ""
            }
        }
    }
}